jwtmiddleware/middleware_test.go

```diff
@@ -86,48 +87,49 @@ func TestSessionFromRequest(t *testing.T) {
 		require.NoError(t, json.NewEncoder(w).Encode(s))
 	})
 	n := negroni.New()
-	n.Use(jwtmiddleware.NewMiddleware(ks, jwtmiddleware.MiddlewareExcludePaths("/anonymous")).NegroniHandler())
+	n.Use(jwtmiddleware.NewMiddleware(ks, jwtmiddleware.MiddlewareExcludePaths("/anonymous")))
 	n.UseHandler(router)
 
 	ts := httptest.NewServer(n)
 	defer ts.Close()
 
 	for k, tc := range []struct {
-		token              string
-		expectedStatusCode int
-		expectedResponse   string
+		token               string
+		expectedStatusCode  int
+		expectedErrorReason string
+		expectedResponse    string
 	}{
 		// token without token
 		{
-			token:              "",
-			expectedStatusCode: 401,
-			expectedResponse:   "Authorization header format must be Bearer {token}",
+			token:               "",
+			expectedStatusCode:  401,
+			expectedErrorReason: "Authorization header format must be Bearer {token}",
 		},
 		// token without kid
 		{
-			token:              "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwiZXhwIjo5OTk5OTk5OTk5LCJzZXNzaW9uIjp7ImlkZW50aXR5Ijp7ImlkIjoiMTIzNDU2Nzg5MCJ9fX0.j0SgjC21nhkNP2QX0uE-I4wDYYRYlZq9wqGeDhrbplkKGW4BOjW5Sk0XFFbqrx68hQYz23QvYOYW5avUBzTjPxHwVqB1HPv6M5P2wHvRn7ZvAyhz83fmJMnBRNBOz1MfjxnEgkwfcVbNqsW2y37kRdZfveBlAzSfuPJV8Rkb4wlBbEGUwoCk78j8zcD_dcYFfXbt7uXz_tscScoIOg959Rmwr2E1XqRNy2qWLKSImwo8athdEEE-byLYytg6mgM02bmEQk2dyd5W2MmqG_4UaiBru6Bf9-drqExHDGUyndnAKi_uvF_131_LkPxy6H5Hu_YfZgSE5hXUbRsBzU-gbY5aV5FSn855PnRDyS_lFnBEn-0vcCIMmxbdfhqyKtFPmFHdSO1YsGruhqYaOLOlTVzThP-1XJSpgMKXHXW35c52zB9AaTV-0ETICvZ_OjZM_uzdWeb6PQmFsztcwdO-9C70yR3_HdcjljvnQ4XHs9ho_3_V57fcbW3uQCTq0TRbwD0AXpkVOvKJqaP1yEXYLKSNpGL2MMkuY-i3k6wTZMTV1280TqbJcSpY5n6WoWJnjoZ08BwBQDfX8AUsKk-D71wJbONqmLo5YnmrS-1gHR3bKCfuUzDdvensLXYJwSHg3ae_qE5VxscRhT_p2odeE8JgQBhd0d6765YBAP93F1c",
-			expectedStatusCode: 401,
-			expectedResponse:   "jwt from authorization HTTP header is missing value for \"kid\" in token header",
```

Comment:
The previous errors exposed here were not really something a legitimate user would run into, and they could not fix it either. The generic version is fine IMO.

Reply:
I kind of disagree, it makes debugging so much more difficult!